server/converters/cxgtool.py

```diff
@@ -385,5 +393,64 @@ def save_metadata(container, metadata):
         A.meta["cxg_properties"] = json.dumps(metadata)
 
 
+def sanitize_keys(keys):
+    """
+    We need names to be safe to use as attribute names in tiledb.  See:
+        TileDB-Inc/TileDB#1575
+        TileDB-Inc/TileDB-Py#294
+    This can be entirely removed once they add proper escaping.
+
+    Args: list of keys
+    Returns: dict of {old_key: new_key, ...}
+
+    Returned new keys will be both safe and unique.
+
+    Masking out [~/.] and anything outside the ASCII range.
+    """
+    # p = re.compile(r"[^a-zA-Z0-9!\-_\.\*'\(\)&$@=;:\+ ,\?]")
+    p = re.compile(r"[^ -\.0-\[\]-\}]")
+    clean_keys = {k: p.sub('_', k) for k in keys}
+
+    used_keys = set()
+    clean_unique_keys = {}
+    for k, v in clean_keys.items():
+        if v not in used_keys:
+            used_keys.add(v)
+            clean_unique_keys[k] = v
+            continue
+
+        # else, needs deduping.
+        counter = 1
```

Comment:
I think you are missing a "counter += 1" inside the while loop.

Reply:
nothing more embarrassing than an infinite loop :-(

Thanks for the catch!